content/tutorials/react-native-presence-publish-subscribe.textile

```diff
@@ -0,0 +1,539 @@
+---
+title: Build a realtime chat app with react native and ably
+section: tutorials
+index: 0
+---
+
+
+h2. Introduction
+
+Messaging apps are on the increase recently. The past few years have brought apps like WhatsApp, Telegram, Facebook Messenger, Slack e.t.c.
+People seem to prefer chat-based applications because they allow for real-time interaction. They also add a personal touch to the experience. 
+In this tutorial, you will learn how to build a real-time chat application using React Native and ably.
+
+React Native lets you build mobile apps using only JavaScript. It uses the same design as React, letting you compose a rich mobile UI from declarative components. To learn more about React Native, please visit "here":https://facebook.github.io/react-native/.
+
+Ably is a real-time data delivery platform providing developers everything they need to create, deliver and manage complex projects. Ably solves the hardest parts so they don’t have to. Our 100% data delivery guarantee offers unique peace of mind to stream data between apps, websites and servers anywhere on earth in milliseconds.
+
+
+h2. Setting Up React Native
+
+To get started first, you need to install the @React Native CLI@ if you don’t already have it installed. To install the @CLI@, run:
+
+```[javascript]
+ npm install -g react-native-cli
+```
+
+After installing the @CLI@, it’s time to create your project. Open a terminal, create a new project called @ably-presence-publish-subscribe-react-native@:
+
+```[javascript]
+react-native init ably-presence-publish-subscribe-react-native
+```
+
+Before I continue, let me explain why I named the project that way.
+
+# ably: I will use the @ably@ real-time library for real-time interactions
+# presence: I will cover the concept of ably presence channels
+# publish: I will cover the concept of publishing messages in ably
+# subscribe: I will cover the concept of subscribing to channels and events in ably
+# react-native: I will build the chat application using @React-Native@ 
+
+Note: You can give your project any other name you wish to.
+
+Wait until React Native does all its installations, then you can change directory into the new project and run the application:
+
+```[sh]
+//change directory to ably-presence-publish-subscribe-react-native
+cd ably-presence-publish-subscribe-react-native
+//run the application for android
+react-native run-android
+//run the application for ios
+react-native run-ios
+```
+
+If all goes well, you should see a screen that says “Welcome to React Native” come up on your emulator or device.
+
+
+h2. Setting Up Ably
+
+You need to sign up on "Ably":https://www.ably.io/, create a new app, click on the @manage app@ button at the left-hand side of the app, click on the @Api keys@  tab and copy our Api Key.
```

Comment:
If you look in any of the other tutorials, you won't need to write this as a common partial is used, see the code `<%= partial 'tutorials/_step-1-setup-free-account' %>`

Reply:
Added this, and taken off the one i did earlier